Parser.java

```diff
@@ -0,0 +1,84 @@
+import java.time.LocalDate;
+import java.util.ArrayList;
+//import java.util.Arrays;
+
+public class Parser {
+
+    /**
+     * Parses a visit string to a Visit instance. Assigns the provided date.
+     * 
+     * @param visit
+     * @param date
+     * @return
+     */
+    static Visit parseVisit(String visit, LocalDate date) {
+        String[] parts = visit.split("\n", 2);
+        String[] name = parts[0].split(",");
+        String notes = parts[1];
+        String[] hashtags = Util.hashtags(notes);
+        Visit visitObj = new Visit(date, name, hashtags, notes);
+        // System.out.println(Arrays.toString(name));
+        // System.out.println(Arrays.toString(hashtags));
+        return visitObj;
+    }
+
+    /**
+     * Parses a trip, representing a date followed by a number of visits, to a list
+     * of visits.
+     * 
+     * @param trip
+     * @return
+     */
+    static ArrayList<Visit> parseTrip(String trip) {
+        // split first line as date
+        // https://stackoverflow.com/questions/23756456/java-separate-string-on-first-line-break
+        String[] parts = trip.split("\n", 2);
+        LocalDate date = Util.parseDate(parts[0].replace("# ", ""));
```

Comment:
This was necessary, as splitting with "\n# " did not remove the initial "# " from the first record.  Maybe a regex like "[\n^]# would have done it, as that means "\n" (newline) OR "^" (beginning of string)